addons/website_event/static/src/interactions/website_event_slot_details.js

```diff
@@ -0,0 +1,123 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+
+const { DateTime } = luxon;
+
+export class SlotDetails extends Interaction {
+    static selector = ".o_wevent_js_slot_details";
+    dynamicSelectors = {
+        ...this.dynamicSelectors,
+        _envBus: () => this.env.bus,
+    };
+    dynamicContent = {
+        _envBus: {
+            "t-on-websiteEvent.enableSubmit": () => this.buttonDisabled = false,
+        },
+        ".a-submit": {
+            "t-on-click.prevent.stop": this._onSubmitClick,
+            "t-att-disabled": () => !this.selectedSlot || this.buttonDisabled ? "disabled" : false,
+        },
+        "li.page-item": {
+            "t-on-click.prevent.stop": this._onChangePageClick,
+        },
+        ".o_wevent_slot_btn": {
+            "t-on-click.prevent.stop": this._onSlotSelected,
+        },
+    };
+
+    get selectedSlot() {
+        return this.form.getAttribute("data-selected-slot-id");
+    }
+
+    setup() {
+        this.currentSlotPage = 0;
+        this.datesLimitPerPage = 6;
+        this.form = this.el.querySelector("#slot_registration_form");
+        this.selectedSlotDatetime = this.el.querySelector(".o_wevent_selected_slot");
+        this.selectedSlotTitle = this.el.querySelector(".o_wevent_selected_slot_title");
+        // Init first slot page
+        if (this.el.querySelector(".pagination")) {
+            this._changePage(0);
+        }
+    }
+
+    /**
+     * Display the specified slot page. Index starting at 0.
+     * @param {Integer} numPage
+     */
+    _changePage(numPage) {
```

Comment:
When trying the next and previous a few times, empty pages will start to appear, and some pages may be repeated?

Reply:
probably a capping issue on the numpage?